### ISSUE TITLE
nginx stats page occasionally not reachable - causes instant switch to offline scene

### ISSUE THREAD

**Issue by happy-dirac** [715209/nginx-obs-automatic-low-bitrate-switching#125]
Hi,

i have observed that the nginx stats page is occasionally not reachable/does not show the values and is just "blank". Don't know why , but had the same when i tried to check it in the browser, it sometimes shows a white page or just the layout with no values. Bitrate is definitely fine, i managed to catch it when I had the liveU stats open and the stream in parallel in VLC, no drops. Switching to a "force-live" scene also works, mobile feed goes on. Bitrate command shows no connection for some time until it goes back to normal.

 Is there a way that NOALBS can ignore this behavior and not switch to the offline scene immediately?

logs when I caught it and forced a live-scene:

2022-10-30T21:18:32.103888Z  INFO NOALBS{user=SuperStreamer}:Switcher: noalbs::switcher: Scene switched to [Offline] offline
2022-10-30T21:18:44.935953Z  INFO noalbs::chat::chat_handler: [SuperStreamer] SuperMod executed command: Switch
2022-10-30T21:18:48.136553Z  INFO NOALBS{user=SuperStreamer}:Switcher: noalbs::switcher: Not able to switch, waiting for scene switch to a switchable scene
2022-10-30T21:20:01.086961Z  INFO noalbs::chat::chat_handler: [SuperStreamer] SuperMod executed command: Bitrate
2022-10-30T21:41:36.716424Z  INFO noalbs::chat::chat_handler: [SuperStreamer] SuperMod  executed command: Bitrate
2022-10-30T21:41:40.117821Z  INFO noalbs::chat::chat_handler: [SuperStreamer] SuperMod  executed command: LiveScene
2022-10-30T21:41:40.148828Z  INFO NOALBS{user=SuperStreamer}:Switcher: noalbs::switcher: Switcher running

when I ignored it (broadcaster was doing sleep stream in a dark room, so i decided to let it run and check how long it takes to fix itself):

2022-10-31T00:19:00.106266Z  INFO NOALBS{user=SuperStreamer}:Switcher: noalbs::switcher: Scene switched to [Offline] offline
2022-10-31T00:20:18.270038Z  INFO NOALBS{user=SuperStreamer}:Switcher: noalbs::switcher: Scene switched to [Normal] mobile-ingest

Anyone having the same issue or knows a fix for this?

Best,

HD

**Comment by DigibitDK** [comment]
You should start using the nginx.conf for NGINX provided by NOALBS, if something is not working correctly ;)

Your NGINX is not set up correctly, and has nothing to do with NOALBS. Change the very first line in nginx-config (worker_processes) back to 1.

/DigibitDK

**Comment by happy-dirac** [comment]
Thanks, it was on auto, i will do a test-run on 1. 

How do I have to understand that as part of the problem? Are the processes totally independent?, 99.9% of the time it works and every once in a while the RNG gods send NOALBS to a different process that has no idea what is going on the other one?

**Comment by DigibitDK** [comment]
Unfortunately I dont know that much about NGINX. If you Google 'NGINX worker_processes', its a bit complicated. At least with RTMP. So, not sure how it works :)

**Comment by happy-dirac** [comment]
Haha, yeah, this is a tough topic to google. Problem seems to be fixed, didn't happen again after i changed it to 1. So thank you again! I thought about it again, and auto/processes not seeing everything should be it. We run everything in docker containers. We only got very sporadic issues in the beginning, but it got way worse after we switched to a machine with twice the core count. So i believe the NGINX container for rtmp got more resources, which made it spawn more processes on auto.
